File: main/diagonal-difference/diagonal-difference.cs
#define DEBUG

using System;
using System.Collections.Generic;
using System.Diagnostics.Contracts;
using System.Linq;

internal static class Solution {
    private static IEnumerable<int> Range(int n)
    {
        return Enumerable.Range(0, n);
    }

    private static int DiagonalDifference(int[][] a)
    {
        var n = a.Length;
        Contract.Assert(a.All(row => row.Length == n));

        var primary = Range(n).Sum(i => a[i][i]);
        var secondary = Range(n).Sum(i => a[i][n - i - 1]);
        return Math.Abs(primary - secondary);
    }

    private static void Main()
    {
        var n = int.Parse(Console.ReadLine());
        var a = new int[n][];

        foreach (var i in Range(n)) {
            var line = Console.ReadLine();
            a[i] = line.Split(' ').Select(int.Parse).Take(n).ToArray();
        }

        Console.WriteLine(DiagonalDifference(a));
    }
}

File: main/diagonal-difference/diagonal-difference.cpp
#include <cstdlib>
#include <iostream>
#include <vector>

namespace {
    std::vector<int> get_row(std::vector<int>::size_type n)
    {
        std::vector<int> row;
        for (row.reserve(n); n != 0u; --n) {
            auto k = 0;
            std::cin >> k;
            row.push_back(k);
        }

        return row;
    }
}

int main()
{
    std::vector<int>::size_type n {};
    std::cin >> n;

    auto sum = 0;
    for (std::vector<int>::size_type i {0u}; i != n; ++i) {
        const auto row = get_row(n);
        sum += row[i] - row[n - i - 1u];
    }

    std::cout << std::abs(sum) << '\n';
}
